packages/playwright-core/src/utils/crypto.ts

```diff
@@ -25,3 +25,179 @@ export function calculateSha1(buffer: Buffer | string): string {
   hash.update(buffer);
   return hash.digest('hex');
 }
+
+const encodeBase128 = (value: number) => {
+  const bytes = new Uint8Array(calculateBase128BytesNeeded(value));
+  const lastPos = bytes.byteLength - 1;
+  let pos = lastPos;
+  do {
+    let byte = value & 0x7f; // Take the last 7 bits
+    value >>>= 7; // Shift right, unsigned
+    if (pos !== lastPos) {
+      byte |= 0x80; // Set the continuation bit on all but the first byte
+    }
+    bytes[pos--] = byte; // Insert the byte at the start of the array
+  } while (value > 0);
+  return bytes;
+};
+
+const calculateBase128BytesNeeded = (num: number) => {
+  // Start at 6 and not 0 to account for overflow and to ensure that the
+  // division below always gives a value equal to or greater than 1.
+  // For example, consider the following 'real' bits needed:
+  // 0: 6 (initial value) + 1 (real) => 7 / 7 = 1
+  // 7: 6 (initial value) + 7 (real) => 13 / 7 = 1
+  // 8: 6 (initial value) + 8 (real) => 14 / 7 = 2
+  let bitsNeeded = 6;
+
+  do {
+    bitsNeeded++;
+    num >>>= 1;
+  } while (num > 0);
+
+  return (bitsNeeded / 7) >>> 0;
+};
+
+class DER {
+  static encodeSequence(data: Buffer[]): Buffer {
+    return this._encode(0x30, Buffer.concat(data));
+  }
+  static encodeInteger(data: number): Buffer {
+    return this._encode(0x02, Buffer.from([data]));
+  }
+  static encodeObject(oid: string): Buffer {
+    const parts = oid.split('.').map((v) => Number(v));
+    // Encode the second part, which could be large, using base-128 encoding if necessary
+    const output = [encodeBase128(40 * parts[0] + parts[1])];
+
+    for (let i = 2; i < parts.length; i++) {
+      output.push(encodeBase128(parts[i]));
+    }
+
+    return this._encode(0x06, Buffer.concat(output));
+  }
+  static encodeNull(): Buffer {
+    return Buffer.from([0x05, 0x00]);
+  }
+  static encodeSet(data: Buffer[]): Buffer {
```

Comment:
I added a comment, didn't find a comment about it in the spec tho. It only says:

> Use a set type to model a collection of variables whose number is known and modest and whose order is insignificant

Reply:
I see the following in `X.6901.doc` section 11.6:

```
The encodings of the component values of a set-of value shall appear in ascending order, the encodings being compared as octet strings with the shorter components being padded at their trailing end with 0-octets. 
NOTE – The padding octets are for comparison purposes only and do not appear in the encodings.
```

Shall we sort things? Or we can assert that `data.length === 1` for now to avoid sorting 😄